esp32c2-hal/Cargo.toml

```diff
@@ -67,9 +67,19 @@ xtal-26mhz           = ["esp-hal-common/xtal-26mhz"]
 xtal-40mhz           = ["esp-hal-common/xtal-40mhz"]
 
 # Embassy support
-embassy              = ["esp-hal-common/embassy"]
-embassy-time-systick = ["esp-hal-common/embassy-time-systick", "embassy-time/tick-hz-16_000_000"]
-embassy-time-timg0   = ["esp-hal-common/embassy-time-timg0",   "embassy-time/tick-hz-1_000_000"]
+embassy                    = ["esp-hal-common/embassy"]
+embassy-time-systick       = ["esp-hal-common/embassy-time-systick", "embassy-time/tick-hz-16_000_000"]
+embassy-time-timg0         = ["esp-hal-common/embassy-time-timg0",   "embassy-time/tick-hz-1_000_000"]
+embassy-executor-interrupt = ["esp-hal-common/embassy-executor-interrupt"]
+embassy-executor-thread    = ["esp-hal-common/embassy-executor-thread"]
+
+# prefer integrated timers, but allow using the generic queue if needed
+embassy-integrated-timers = ["esp-hal-common/embassy-integrated-timers"]
+embassy-generic-queue-8   = ["esp-hal-common/embassy-generic-queue-8"]
```

Comment:
We don't need to provide these features, though we could turn off code in the timer drivers if `integrated-timers` isn't enabled. And at that point providing `generic-queue-*` is just there for a bit of consistency. Not sure if I should keep these.

Reply:
My main worry is that if we rely on the user to set stuff on `embassy-executor` they will bump up against the existing riscv32 stuff in there and end up confused.